### PR TITLE
[Doc] Update PR checklist

### PR DESCRIPTION
**Pull request by waffle87**
## Description

Add some more things that are now required in `info.json` and specify some other things that should *not* be included (let me know if I've missed any) and other misc clean up.

## Types of Changes

- [x] Documentation

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by elpekenin** [comment]
Even tho the file is named `pr_checklist.md`(singular), the very first heading is plural: `# PR checklists`.
No idea if it is kept like this for some reason (eg: not breaking some link), I'm pointing that small inconsistency just in case it was missed.

File: docs/pr_checklist.md [hunk 1]
```diff
@@ -69,15 +69,25 @@ https://github.com/qmk/qmk_firmware/pulls?q=is%3Apr+is%3Aclosed+label%3Akeyboard
...
+    - must follow the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md)
     - flash command is present, and has `:flash` at end
     - valid hardware availability link (unless handwired) -- private groupbuys are okay, but one-off prototypes will be questioned. If open-source, a link to files should be provided.
     - clear instructions on how to reset the board into bootloader mode
     - a picture about the keyboard and preferably about the PCB, too
         - images are not to be placed in the `qmk_firmware` repository
         - images should be uploaded to an external image hosting service, such as [imgur](https://imgur.com/).
         - if imgur is used, images should be resized appropriately: append "h" to the image url i.e. `https://i.imgur.com/vqgE7Ok.jpg` becomes `https://i.imgur.com/vqgE7Okh.jpg`
+        - image links should link directly to the image, not a "preview" (e.g. [https://imgur.com/vqgE7Ok](https://imgur.com/vqgE7Ok) should be [https://i.imgur.com/vqgE7Okh.jpg](https://i.imgur.com/vqgE7Okh.jpg) when using imgur)
```
Comment (elpekenin) [inline docs/pr_checklist.md:90]:
```suggestion
        - image links should link directly to the image, not a "preview", e.g. append the file extension to the image url when using imgur: [https://imgur.com/vqgE7Ok](https://imgur.com/vqgE7Ok) becomes [https://i.imgur.com/vqgE7Okh.jpg](https://i.imgur.com/vqgE7Okh.jpg)
```

**Review by elpekenin — COMMENTED** [review]
Hopefully makes the test easier to understand + make both lines consistent

File: docs/pr_checklist.md [hunk 2]
```diff
@@ -69,15 +69,25 @@ https://github.com/qmk/qmk_firmware/pulls?q=is%3Apr+is%3Aclosed+label%3Akeyboard
...
 - `readme.md`
-    - standard template should be present -- [link to template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md)
+    - must follow the [template](https://github.com/qmk/qmk_firmware/blob/master/data/templates/keyboard/readme.md)
     - flash command is present, and has `:flash` at end
     - valid hardware availability link (unless handwired) -- private groupbuys are okay, but one-off prototypes will be questioned. If open-source, a link to files should be provided.
     - clear instructions on how to reset the board into bootloader mode
     - a picture about the keyboard and preferably about the PCB, too
         - images are not to be placed in the `qmk_firmware` repository
         - images should be uploaded to an external image hosting service, such as [imgur](https://imgur.com/).
         - if imgur is used, images should be resized appropriately: append "h" to the image url i.e. `https://i.imgur.com/vqgE7Ok.jpg` becomes `https://i.imgur.com/vqgE7Okh.jpg`
```
Comment (elpekenin) [inline docs/pr_checklist.md:89]:
```suggestion
        - if imgur is used, images should be resized appropriately, i.e. append "h" to the image url: [https://i.imgur.com/vqgE7Ok.jpg](https://i.imgur.com/vqgE7Ok.jpg) becomes [https://i.imgur.com/vqgE7Okh.jpg](https://i.imgur.com/vqgE7Okh.jpg)
```

Comment (elpekenin) [inline docs/pr_checklist.md:89]:
```suggestion
        - if imgur is used, images should be resized appropriately, i.e. append "h" to the image url: [https://i.imgur.com/vqgE7Ok.jpg](https://i.imgur.com/vqgE7Ok.jpg) becomes [https://i.imgur.com/vqgE7Okh.jpg](https://i.imgur.com/vqgE7Okh.jpg)
```

**Review by drashna — APPROVED** [review]

**Pull request merged**